### PR TITLE
Document skipBom in CharReaderBuilder

### PR DESCRIPTION
**Pull request by axic**
This was introduced in #1149 (and renamed in #1162) but as far as I can tell it was never documented.

### CODE REVIEW

File: include/json/reader.h [hunk 1]
```diff
@@ -324,6 +324,8 @@ class JSON_API CharReaderBuilder : public CharReader::Factory {
    * - `"allowSpecialFloats": false or true`
    *   - If true, special float values (NaNs and infinities) are allowed and
    *     their values are lossfree restorable.
+   * - `"skipBom": false or true`
+   *   - If true, if the input starts with the Unicode BOM, it is skipped.
```
Comment (BillyDonahue) [inline include/json/reader.h:328]:
```suggestion
   *   - If true, if the input starts with the Unicode byte order mark, it is skipped.
```

**Review by BillyDonahue — APPROVED** [review]

File: include/json/reader.h (hunk 1 above)
Comment (axic) [reply include/json/reader.h]: Updated it with `byte order mark (BOM)` if that's okay?

Comment (BillyDonahue) [reply include/json/reader.h]: ok

File: include/json/reader.h [hunk 2]
```diff
@@ -324,6 +324,8 @@ class JSON_API CharReaderBuilder : public CharReader::Factory {
    * - `"allowSpecialFloats": false or true`
    *   - If true, special float values (NaNs and infinities) are allowed and
    *     their values are lossfree restorable.
+   * - `"skipBom": false or true`
+   *   - If true, if the input starts with the Unicode byte order mark (BOM), it is skipped.
```
Comment (BillyDonahue) [inline include/json/reader.h:328]: This line is wider than the 80-column convention. We can take out a little redundancy and shorten it, or wrap before 80 columns.

```suggestion
   *   - If true, skip over any leading Unicode byte order mark (BOM).
```

**Review by BillyDonahue — CHANGES_REQUESTED** [review]

File: include/json/reader.h [hunk 3]
```diff
@@ -324,6 +324,9 @@ class JSON_API CharReaderBuilder : public CharReader::Factory {
    * - `"allowSpecialFloats": false or true`
    *   - If true, special float values (NaNs and infinities) are allowed and
    *     their values are lossfree restorable.
+   * - `"skipBom": false or true`
+   *   - If true, if the input starts with the Unicode byte order mark (BOM),
+         it is skipped.
```
Comment (BillyDonahue) [inline include/json/reader.h:329]:
```suggestion
  *      it is skipped.
```

**Review by BillyDonahue — COMMENTED** [review]
asterisk

File: include/json/reader.h [hunk 4]
```diff
@@ -324,6 +324,9 @@ class JSON_API CharReaderBuilder : public CharReader::Factory {
    * - `"allowSpecialFloats": false or true`
    *   - If true, special float values (NaNs and infinities) are allowed and
    *     their values are lossfree restorable.
+   * - `"skipBom": false or true`
+   *   - If true, if the input starts with the Unicode byte order mark (BOM),
+  *      it is skipped.
```
Comment (BillyDonahue) [inline include/json/reader.h:329]:
```suggestion
   *     it is skipped.
```

**Review by BillyDonahue — COMMENTED** [review]
adjust

**Comment by axic** [comment]
Thanks!

**Pull request merged**